gatherers/service-worker.js

```diff
@@ -20,11 +20,15 @@ const Gather = require('./gather');
 
 class ServiceWorker extends Gather {
   setup(options) {
+    this.resolved = false;
+
     this.artifactsResolved = new Promise((res, _) => {
       options.driver.on(
           'ServiceWorker.workerVersionUpdated', data => {
-            if (ServiceWorker.getActivatedServiceWorker(data.versions) !== undefined) {
+            if (ServiceWorker.getActivatedServiceWorker(data.versions) !== undefined &&
+                !this.resolved) {
               this.artifact = {serviceWorkers: data};
+              this.resolved = true;
```

Comment:
The issue is,

we resolve the promise like all the work is done however 

`'ServiceWorker.workerVersionUpdated', data => {`

keeps firing over and over again... changing the value of this.artifact which is problematic as we do not interact with the this.artifact value until potentially a while after the promise was resolved. Thus opening the door for mutation before getting the state of the value.

This guard clause is to prevent that mutation from happening.